src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -123,199 +122,206 @@
+                                                     {:content (fn []
+                                                                 [select-asset-bottom-sheet
+                                                                  clear-input!])}])
+        sender-network-values                     (rf/sub
+                                                   [:wallet/wallet-send-sender-network-values])
+        receiver-network-values                   (rf/sub
+                                                   [:wallet/wallet-send-receiver-network-values])
+        token-not-supported-in-receiver-networks? (every? #(= (:type %) :not-available)
+                                                          (filter #(not= (:type %) :add)
+                                                                  receiver-network-values))
+        suggested-routes                          (rf/sub [:wallet/wallet-send-suggested-routes])
+        routes                                    (when suggested-routes
+                                                    (or (:best suggested-routes) []))
+        no-routes-found?                          (and
+                                                   (every-network-value-is-zero?
+                                                    sender-network-values)
+                                                   (not (nil? routes))
+                                                   (not loading-routes?)
+                                                   (not token-not-supported-in-receiver-networks?))
+        receiver-networks                         (rf/sub [:wallet/wallet-send-receiver-networks])
+        receiver-preferred-networks               (rf/sub
+                                                   [:wallet/wallet-send-receiver-preferred-networks])
+        receiver-preferred-networks-set           (set receiver-preferred-networks)
+        sending-to-unpreferred-networks?          (not (every? (fn [receiver-selected-network]
+                                                                 (contains?
+                                                                  receiver-preferred-networks-set
+                                                                  receiver-selected-network))
+                                                               receiver-networks))
+        current-address                           (rf/sub [:wallet/current-viewing-account-address])
+       ]
+    (rn/use-mount
+     (fn []
+       (let [dismiss-keyboard-fn   #(when (= % "active") (rn/dismiss-keyboard!))
+             app-keyboard-listener (.addEventListener rn/app-state "change" dismiss-keyboard-fn)]
+         #(.remove app-keyboard-listener))))
+    (rn/use-effect
+     (fn []
+       (set-input-state #(controlled-input/set-upper-limit % current-limit)))
+     [current-limit])
+
+    (rn/use-effect
+     (fn []
+       (set-input-state controlled-input/init-state))
+     [current-address])
```

Comment:
This hook flushes the input to it's default state once the current address changes

Reply:
checks it is not the default value as otherwise it causes a bug on the initial render. Only want this to fire if the account has changed and the input is not the initial state.

Reply:
I know when `current-address` changes, the values (`input-state`) in the callback would also be updated. But I think for correctness, we can use the value for `input-state` from the callback for setState in `set-input-state`.

That would be something like this:

```cljs
(rn/use-effect
         (fn []
           (set-input-state #(if (= % controlled-input/init-state)
                               %
                               controlled-input/init-state)))
         [current-address])
```

What do you think?

Reply:
Hey @J-Son89, what happens if we have routes shown on the screen, and we change the account, are routes cleaned too?
If not, probably we should call `(rf/dispatch [:wallet/clean-suggested-routes])` here too.

Reply:
Thanks @briansztamfater gonna check this!

Reply:
Just noticed this while testing in one of my rebased branches:
If user disables a network and then the account changes, the disabled networks are not reset. I think we should reset disabled networks as well because the switched account might have balance only on the previously disabled networks, leading the user to a dead-end.
For that, we should call (rf/dispatch [:wallet/clean-disabled-from-networks]) here too. We can raise a new PR for this, let me know how would you like to handle it @J-Son89!